runtime/libc/wasi_sdk_backing.c

```diff
@@ -27,22 +27,55 @@
 #define fdatasync fsync
 #endif
 
-int main(int argc, char* argv[]) {
-    runtime_main(argc, argv);
+// Code that actually runs the wasm code
+IMPORT void wasmf__start(void);
+u32 runtime_argc = 0;
+char *runtime_argv_buffer = NULL;
+u32 runtime_argv_buffer_len = 0;
+u32 *runtime_argv_buffer_offsets = NULL;
+
+void runtime_cleanup() {
+    // TODO: What if atexit triggers when inside of runtime?
+    // Before implementing more complex state machine transitions, we should decide if segmentation
+    // (the only thing that uses these calls) merits this additional complexity.
+    switch_into_runtime();
```

Comment:
Shouldn't we already be in runtime? This is called after `main` returns.

I don't have strong opinions on if we should keep it or not.